src/TalisOrm/AggregateRepository.php

```diff
@@ -237,4 +248,61 @@ private function select($selectExpression, $tableExpression, array $where)
 
         return $this->connection->executeQuery($sql, $values);
     }
+
+    private function rememberEntity(Entity $entity)
+    {
+        $this->knownEntities[spl_object_hash($entity)] = $entity;
+    }
+
+    /**
+     * @param Entity $entity
+     * @return bool
+     */
+    private function exists(Entity $entity)
```

Comment:
The `exists` name here is misleading, as it doesn't really indicate whether the aggregate _exists_, but only if the repository _knows_ about it, which are entirely different things.